From: Peter Kokot <peterkokot@gmail.com>
Subject: Fix tests

- ext/zip
  When PHP is compiled and built agains libzip that doesn't have AES
  support enabled (if all encryption options are set to OFF in CMake),
  ZipArchive::EM_AES_256 and ZipArchive::EM_AES_192 aren't supported.
---
 ext/zip/tests/oo_addglob2.phpt   | 6 ++++++
 ext/zip/tests/oo_encryption.phpt | 6 ++++++
 2 files changed, 12 insertions(+)

diff --git a/ext/zip/tests/oo_addglob2.phpt b/ext/zip/tests/oo_addglob2.phpt
index 517c0b7fd7f..1432d704de5 100644
--- a/ext/zip/tests/oo_addglob2.phpt
+++ b/ext/zip/tests/oo_addglob2.phpt
@@ -6,6 +6,12 @@
 <?php
 if (!method_exists('ZipArchive', 'setEncryptionName')) die('skip encrytion not supported');
 if(!defined("GLOB_BRACE")) die ('skip requires GLOB_BRACE');
+if (
+    method_exists('ZipArchive', 'isEncryptionMethodSupported')
+    && !ZipArchive::isEncryptionMethodSupported(ZipArchive::EM_AES_256)
+) {
+    die('skip EM_AES_256 encryption not supported');
+}
 ?>
 --FILE--
 <?php
diff --git a/ext/zip/tests/oo_encryption.phpt b/ext/zip/tests/oo_encryption.phpt
index f5207e30759..ccc0a04b8f1 100644
--- a/ext/zip/tests/oo_encryption.phpt
+++ b/ext/zip/tests/oo_encryption.phpt
@@ -5,6 +5,12 @@
 --SKIPIF--
 <?php
 if (!method_exists('ZipArchive', 'setEncryptionName')) die('skip encryption not supported');
+if (
+    method_exists('ZipArchive', 'isEncryptionMethodSupported')
+    && !ZipArchive::isEncryptionMethodSupported(ZipArchive::EM_AES_256)
+) {
+    die('skip EM_AES_256 encryption not supported');
+}
 ?>
 --FILE--
 <?php
